bin/cancel-all-pending-txs.js

```diff
@@ -0,0 +1,226 @@
+  fetchRequest.setHeader('Authorization', `Bearer ${GLIF_TOKEN}`)
+  const provider = new ethers.JsonRpcProvider(fetchRequest, null, {
+    polling: true,
+    batchMaxCount: 10
+  })
+
+  const signer = ethers.Wallet.fromPhrase(WALLET_SEED, provider)
+  const walletDelegatedAddress = newDelegatedEthAddress(/** @type {any} */(signer.address), CoinType.MAIN).toString()
+  console.log(
+    'Wallet address:',
+    signer.address,
+    walletDelegatedAddress
+  )
+
+  return { provider, signer, walletDelegatedAddress }
+}
+
+/**
+ * @returns {Promise<{
+   "cid": string;
+   "nonce": number;
+   "height": number;
+   "timestamp": number;
+   "gasLimit": number;
+   "gasFeeCap": string;
+   "gasPremium": string;
+   "method": string;
+   "methodNumber": number;
+   "receipt": {
+      "exitCode": number;
+      "return": string;
+      "gasUsed": number;
+    },
+    "size": number;
+    "error": string;
+    "baseFee": string;
+    "fee": {
+      "baseFeeBurn": string;
+      "overEstimationBurn": string;
+      "minerPenalty": string;
+      "minerTip": string;
+      "refund": string;
+    },
+  }>}
+ */
+async function getRecentSendMessage () {
+  let res = await fetch('https://filfox.info/api/v1/message/list?method=Send')
+  if (!res.ok) {
+    throw new Error(`Filfox request failed with ${res.status}: ${(await res.text()).trimEnd()}`)
+  }
+  const body = /** @type {any} */(await res.json())
+  assert(body.messages.length > 0, '/message/list returned an empty list')
+  const sendMsg = body.messages.find(m => m.method === 'Send')
+  assert(!!sendMsg, 'No Send message found in the recent committed messages')
+  const cid = sendMsg.cid
+
+  res = await fetch(`https://filfox.info/api/v1/message/${cid}`)
+  if (!res.ok) {
+    throw new Error(`Filfox request failed with ${res.status}: ${(await res.text()).trimEnd()}`)
+  }
+
+  return /** @type {any} */(await res.json())
+}
+
+/**
+ * @param {ReturnType<getRecentSendMessage>} recentSendMessage
+ * @param {{
+ *  cid: string;
+ *  gasPremium: string;
+ *  nonce: number;
+ *  createdAt: Date;
+ * }} tx
+ */
+async function cancelTransaction (recentSendMessage, tx) {
+  // This would skip replacement TXs we created by this script. That's not what we want.
+  // if (tx.createdAt.getTime() > Date.now() - 30 * 60_000) {
+  //   console.log('TX %s (nonce %s) was created at %s (less than 30 minutes ago), will not cancel it', tx.cid, tx.nonce, tx.createdAt)
+  //   return
+  // }
+
+  console.log('REPLACING %s (nonce %s, created at %s)', tx.cid, tx.nonce, tx.createdAt)
+
+  const gasUsed = recentSendMessage.receipt.gasUsed
+  const gasFeeCap = Number(recentSendMessage.gasFeeCap)
+  const oldGasPremium = tx.gasPremium
+  const nonce = tx.nonce
+
+  console.log(' - SENDING THE REPLACEMENT TRANSACTION')
+  try {
+    const replacementTx = await signer.sendTransaction({
+      to: signer.address,
+      value: 0,
+      nonce,
+      gasLimit: Math.ceil(gasUsed * 1.1),
+      maxFeePerGas: gasFeeCap,
```

Comment:
This patch is missing: https://github.com/filecoin-station/spark-evaluate/blob/e4358f466c53c5f457c4afcade70b3f47a95c128/lib/cancel-stuck-txs.js#L111-L112

Reply:
I'll fix that in a follow-up PR.

Reply:
https://github.com/filecoin-station/utils/pull/2